src/ERC3156FlashLoanSolverWrapper.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8;
+
+import {ICowSettlement} from "./interface/ICowSettlement.sol";
+import {TransientStorageArray} from "./mixin/TransientStorageArray.sol";
+import {ICowAuthentication} from "./vendored/ICowAuthentication.sol";
+import {IERC20} from "./vendored/IERC20.sol";
+import {IERC3156FlashBorrower} from "./vendored/IERC3156FlashBorrower.sol";
+import {IERC3156FlashLender} from "./vendored/IERC3156FlashLender.sol";
+
+contract ERC3156FlashLoanSolverWrapper is IERC3156FlashBorrower, TransientStorageArray {
+    struct LoanRequest {
+        IERC20 token;
+        uint256 amount;
+    }
+
+    /// @notice ERC 3156 requires flash loan lenders to return this value if
+    /// execution was successful.
+    bytes32 private constant ERC3156_ONFLASHLOAN_SUCCESS = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    /// @notice The contract that will be called when the flash-loan contract
+    /// calls back to this contract.
+    ICowSettlement public immutable settlementContract;
+    ICowAuthentication public immutable settlementAuthentication;
+
+    modifier onlySettlementContract() {
+        require(msg.sender == address(settlementContract), "Only callable in a settlement");
+        _;
+    }
+
+    modifier onlySolver() {
+        // Todo: investigate security implication of self calls.
+        require(settlementAuthentication.isSolver(msg.sender), "Only callable by a solver");
+        _;
+    }
+
+    constructor(ICowSettlement _settlementContract) {
+        settlementContract = _settlementContract;
+        settlementAuthentication = ICowAuthentication(_settlementContract.authenticator());
+    }
+
+    function flashLoanAndSettle(IERC3156FlashLender lender, LoanRequest calldata loan, bytes calldata settlement)
```

Comment:
Is there a way of how we could allow for multiple loans for a single settlement (e.g. if two orders that require flash loans ought to be settled)?